### ISSUE TITLE
Permission denied when API is trying to create extracted-resources path in docker env

### ISSUE THREAD

**Issue by newtewt** [medic/medic#5519]
Deploying master to gamma.dev is crashing the instance. The error is shown below. Looks like we are trying to create a new directory and we don't have enough permissions. 

The line causing the crash. 

https://github.com/medic/medic/blob/868a6402523339e7e7b8086617bd1a9adc96ba00/api/src/resource-extraction.js#L20

The error

```
[2019-03-19 13:51:58] 2019-03-19 13:51:58 ERROR: Fatal error initialising medic-api 
[2019-03-19 13:51:58] 2019-03-19 13:51:58 ERROR: { Error: EACCES: permission denied, mkdir '/srv/software/medic-api/md5-XlLcXD+0QO4uCjc5M7rmw==/src/extracted-resources/'
[2019-03-19 13:51:58]     at Object.fs.mkdirSync (fs.js:885:18)
[2019-03-19 13:51:58]     at createFolderIfDne (/srv/software/medic-api/md5-XlLcXD+0QO4uCjc5M7rmw==/src/resource-extraction.js:20:56)
[2019-03-19 13:51:58]     at extractCacheableAttachments (/srv/software/medic-api/md5-XlLcXD+0QO4uCjc5M7rmw==/src/resource-extraction.js:23:3)
[2019-03-19 13:51:58]     at <anonymous>
[2019-03-19 13:51:58]     at process._tickCallback (internal/process/next_tick.js:188:7)
[2019-03-19 13:51:58]   [stack]: 'Error: EACCES: permission denied, mkdir \'/srv/software/medic-api/md5-XlLcXD+0QO4uCjc5M7rmw==/src/extracted-resources/\'\n    at Object.fs.mkdirSync (fs.js:885:18)\n    at createFolderIfDne (/srv/software/medic-api/md5-XlLcXD+0QO4uCjc5M7rmw==/src/resource-extraction.js:20:56)\n    at extractCacheableAttachments (/srv/software/medic-api/md5-XlLcXD+0QO4uCjc5M7rmw==/src/resource-extraction.js:23:3)\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)',
[2019-03-19 13:51:58]   [message]: 'EACCES: permission denied, mkdir \'/srv/software/medic-api/md5-XlLcXD+0QO4uCjc5M7rmw==/src/extracted-resources/\'',
[2019-03-19 13:51:58]   errno: -13,
[2019-03-19 13:51:58]   code: 'EACCES',
[2019-03-19 13:51:58]   syscall: 'mkdir',
[2019-03-19 13:51:58]   path: '/srv/software/medic-api/md5-XlLcXD+0QO4uCjc5M7rmw==/src/extracted-resources/' } 
```

**Steps to reproduce**:
Log in to gamma.dev as admin
Navigate to upgrade page.
Apply master branch.

**What should happen**:
Master branch is deployed without issue. 

**What actually happens**:
App is crashing. 

- Instance: gamma.dev
- Browser: chrome
- Client platform: linux
- App: webapp
- Version: master/3.5ish

**Comment by kennsippell** [comment]
@Hareet We need API to make a new folder. It is used to serve static resources for unauthenticated users. We create the folder when API starts up.

Do you have guidance on where to put this inside the docker container? Should I be putting this folder somewhere else?

If the path we're using above is as good as any place, do you think we can solve this with a file permission change inside the docker container?

**Comment by garethbowen** [comment]
@Hareet @henokgetachew Do you have any advice here?

**Comment by SCdF** [comment]
For testing, you can work around this by doing:

```
`chmod 777 -R /srv/`
```
And restarting API.

Obviously not a PROD-ready solution.

**Comment by Hareet** [comment]
@kennsippell 

Can you move this folder creation to `/tmp` ? It's safer, not on persistent storage, so if container restarts, API will regenerate files.

In medic-os: `horticulturalist` user unzips a medic-api version onto disk. The `horti` user owns writes and `medic-api` user can serve.

**Comment by Hareet** [comment]
Due to a large number of projects already existing on 3.x, back-porting an environment variable to hold the directory value `/tmp` is cumbersome. 

The way medic-os is constructed as a monolith container that writes software packages and its data store to an EBS volume limits how quickly backporting is executed. An EBS volume can only be mounted to one server at a time, so you can not roll the docker deployment resource without unmounting the volume, and fighting the built-in race condition of auto-mounting it back on a new server. This issue is alleviated when medic-os is separated into service containers (API, Sentinel, Horti, couchDB), which leads CouchDB to be the only container requiring EBS, and that will be separated in a different pod so the other services can upgrade/back-port/change and roll deployments as required.

tl;dr - Our current setup means its uglier to backport an environment variable than to request devs to hardcode temporarily to `/tmp`

@kennsippell Let me know if you want more details

**Comment by garethbowen** [comment]
@kennsippell Do you have time to look at this issue this week?

**Comment by newtewt** [comment]
Confirmed no crashing and resources are stored in tmp. Tested on local and gamma.dev
